src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -68,6 +68,7 @@ export class QueryStringManager {
     const query = {
       query: defaultQuery,
       language: defaultLanguageId,
+      dataset: undefined,
```

Comment:
why is this necessary? i see query is used to construct `newQuery` and dataset is set to be defaultDataset in line 81

Reply:
my linter was complaining because i thought the return type was just query language